src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -0,0 +1,560 @@
+                                       shiftEnabled,
+                                       isOnOriginalPosition,
+                                       _selectionNeedsToBeCopied);
+
+            if (_core->HasSelection())
+            {
+                // GH#9787: if selection is active we don't want to track the touchdown position
+                // so that dragging the mouse will extend the selection rather than starting the new one
+                _singleClickTouchdownPos = std::nullopt;
+            }
+        }
+        else if (buttonState.isRightButtonDown)
+        {
+            // CopyOnSelect right click always pastes
+            if (_core->CopyOnSelect() || !_core->HasSelection())
+            {
+                PasteTextFromClipboard();
+            }
+            else
+            {
+                CopySelectionToClipboard(shiftEnabled, nullptr);
+            }
+        }
+    }
+
+    void ControlInteractivity::Touched(const winrt::Windows::Foundation::Point contactPoint)
+    {
+        _touchAnchor = contactPoint;
+    }
+
+    void ControlInteractivity::PointerMoved(const winrt::Windows::Foundation::Point mouseCursorPosition,
+                                            TerminalInput::MouseButtonState buttonState,
+                                            const unsigned int pointerUpdateKind,
+                                            const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                                            const bool focused,
+                                            const til::point terminalPosition)
+    {
+        // Short-circuit isReadOnly check to avoid warning dialog
+        if (focused && !_core->IsInReadOnlyMode() && _canSendVTMouseInput(modifiers))
+        {
+            _trySendMouseEvent(pointerUpdateKind, buttonState, modifiers, terminalPosition);
+        }
+        else if (focused && buttonState.isLeftButtonDown)
+        {
+            if (_singleClickTouchdownPos)
+            {
+                // Figure out if the user's moved a quarter of a cell's smaller axis away from the clickdown point
+                auto& touchdownPoint{ *_singleClickTouchdownPos };
+                auto distance{ std::sqrtf(std::powf(mouseCursorPosition.X - touchdownPoint.X, 2) +
+                                          std::powf(mouseCursorPosition.Y - touchdownPoint.Y, 2)) };
+
+                const auto fontSizeInDips{ _core->FontSizeInDips() };
+                if (distance >= (std::min(fontSizeInDips.width(), fontSizeInDips.height()) / 4.f))
+                {
+                    _core->SetSelectionAnchor(terminalPosition);
+                    // stop tracking the touchdown point
+                    _singleClickTouchdownPos = std::nullopt;
+                    _singleClickTouchdownTerminalPos = std::nullopt;
+                }
+            }
+
+            SetEndSelectionPoint(terminalPosition);
+
+            // TODO: Make sure this still works

```

Comment:
Well?

Reply:
YEAH does it

Reply:
Wait... so what handles all of this now?

Reply:
this is still in TermControl - I just had copied that body, pasted in interactivity, then commented that bit out and left it behind in TermControl. Still works like a charm